### PR TITLE
🚀 Add affinity and tolerations to helm chart

### PR DESCRIPTION
**Pull request by Pale-whale**
### Description

This pr adds the ability to modify the affinity and tolerations of the operator

### Usage Example

```
operator:
  tolerations:
  - key: "kubernetes.azure.com/scalesetpriority"
    operator: "Equal"
    value: "spot"
    effect: "NoSchedule"
  affinity:
    nodeAffinity:
    requiredDuringSchedulingIgnoredDuringExecution:
      nodeSelectorTerms:
      - matchExpressions:
        - key: kubernetes.io/arch
          operator: In
          values:
          - amd64
```

### Community Note

* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request.
* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### CODE REVIEW

**Comment by arybolovlev** [comment]
Hi @Pale-whale,

Thank you for your interest in contributing to this project. I am going to review this PR next week. In the meantime, could you please sign the CLA(Contributor License Agreement) that is pending?

Thanks!

**Comment by arybolovlev** [comment]
Hi @Pale-whale,

Thank you again for your contribution, there are a few things that I would like to ask you to add here and we should be good to go.

- Let's move both `affinity` and `tolerations` to the root values level, as it is done with `imagePullSecrets` and `replicaCount`. Those are the 'global' options whereas those that go under the `operator` are either container or binary-specific.
- Please add both `affinity` and `tolerations` to the `values.yaml` file with their default values, for example,`affinity: {}`.
- Please add documentation string and usage examples for both `affinity` and `tolerations`. Let's keep the usage example away from the description and make it go first. We use [helm-docs](https://github.com/norwoodj/helm-docs) to generate Helm chart readme file, so an example would be the following:
  ```yaml
  # Usage example:
  # affinity:
  #   nodeAffinity:
  #     requiredDuringSchedulingIgnoredDuringExecution:
  #       nodeSelectorTerms:
  #       - matchExpressions:
  #         - key: kubernetes.io/arch
  #           operator: In
  #           values:
  #         - amd64
  #
  # -- Kubernetes Affinity.
  # More information: https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity
  ```
- Once you added the doc strings, please run `make helm-docs` and a new readme file will be generated.
- Please add a [change log entry](https://github.com/hashicorp/terraform-cloud-operator/blob/main/CONTRIBUTING.md#write-a-changelog):
  ```console
  $ changie new --kind ENHANCEMENTS --custom "PR=350" --body '`Helm Chart`: Add new attributes `affinity` and `tolerations` to specify how operator pods should be scheduled on nodes in a cluster.'
  ```

Thank you!

**Comment by Pale-whale** [comment]
Hi @arybolovlev,

Thanks for the insights, let me know if anything else is needed.

Thank you 😃

File: charts/terraform-cloud-operator/values.yaml [hunk 1]
```diff
@@ -4,6 +4,32 @@
...
+# Usage example:
+# affinity:
+#   nodeAffinity:
+#     requiredDuringSchedulingIgnoredDuringExecution:
+#       nodeSelectorTerms:
+#       - matchExpressions:
+#         - key: kubernetes.io/arch
+#           operator: In
+#           values:
+#         - amd64
```
Comment (arybolovlev) [inline charts/terraform-cloud-operator/values.yaml:16]:
```suggestion
#           - amd64
```

File: charts/terraform-cloud-operator/values.yaml [hunk 2]
```diff
@@ -4,6 +4,32 @@
...
+# Usage example:
+# tolerations:
+# - key: "kubernetes.azure.com/scalesetpriority"
+#   operator: "Equal"
+#   value: "spot"
+#   effect: "NoSchedule"
+#
+# -- Kubernetes Tolerations.
+# More information: https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/
+tolerations: {}
```
Comment (arybolovlev) [inline charts/terraform-cloud-operator/values.yaml:31]:
```suggestion
tolerations: []
```

**Review by arybolovlev — CHANGES_REQUESTED** [review]
All it all looks great! Two small suggestions:

- Fix an example usage
- Make `tolerations` a list

Please, don't forget to run `make helm-docs` to apply changes to the docs. Thanks!

**Comment by arybolovlev** [comment]
Hi @Pale-whale,

Please let me know if you can make the requested changes.

Thanks.

File: charts/terraform-cloud-operator/values.yaml (hunk 1 above)
Comment (arybolovlev) [inline charts/terraform-cloud-operator/values.yaml:16]:
```suggestion
#           - amd64
```

File: charts/terraform-cloud-operator/values.yaml (hunk 2 above)
Comment (arybolovlev) [inline charts/terraform-cloud-operator/values.yaml:31]:
```suggestion
tolerations: []
```

**Pull request merged**